app/assets/javascripts/discourse/app/components/reviewable-item.js

```diff
@@ -225,6 +228,35 @@ export default Component.extend({
     this._penalize("showSilenceModal", reviewable, performAction);
   },
 
+  clientEdit(reviewable, performAction) {
+    this.store.find("post", reviewable.post_id).then((p) => {
+      Topic.find(p.topic_id, {}).then((t_json) => {
+        const t = Topic.create(t_json);
+        p.set("topic", t);
+
+        if (!this.currentUser) {
+          return this.dialog.alert(I18n.t("post.controls.edit_anonymous"));
+        } else if (!p.can_edit) {
+          return false;
+        }
+
+        const opts = {
+          post: p,
+          action: Composer.EDIT,
+          draftKey: p.get("topic.draft_key"),
+          draftSequence: p.get("topic.draft_sequence"),
+          skipDraftCheck: true,
+          skipJumpOnSave: true,
+        };
+
+        this.composer.open(opts);
+
+        return performAction();
+      });
+    });
+    return;
```

Comment:
what is the goal of this return?

Reply:
believe the customer plugin this was pulled from needed it at one point. Removed.